src/main/java/domain/LottoNumber.java

```diff
@@ -0,0 +1,50 @@
+package domain;
+
+import static validator.NumberValidators.validateLottoNumberRange;
+
+import java.util.HashMap;
+
+public class LottoNumber implements Comparable<LottoNumber> {
+
+    private static final HashMap<Integer, LottoNumber> cache = new HashMap<>();
+
+    private final int number;
+
+    private LottoNumber(int value) {
+        number = value;
+    }
+
+    public static LottoNumber of(int value) {
+        LottoNumber lottoNumber = getCacheIfExists(value);
+
+        if (lottoNumber != null) {
+            return lottoNumber;
+        }
+
+        validateLottoNumberRange(value);
+        return createNewCache(value);
+    }
```

Comment:
> 여러곳에서 사용되는 로또숫자들을 LottoNumber라는 Value Object를 만들었습니다.
> 페어가 블로그에서 발견한 캐슁 관련 코드를 공유해주어 보았는데, �LottoNumber라는 클래스 내부에 클래스 변수로 LottoNumber 인스턴스들을 캐쉬로 관리하고 있었습니다.
> 저 같은 경우 예전에 데이터를 담을 Cache와 캐쉬를 담을 CacheRepository를 별도로 만들어서 관리한 적이 있는데, 시야를 넓혀보고자 한번 이 방식을 참고해보았습니다.
> 다만, 해당 예시 코드에서는 static 블록 내부에서 프로그램 시작시 모든 LottoNumber 캐쉬를 생성하였는데, 저는 해당 번호가 실제로 필요해지는 첫 시점에 캐쉬를 생성하도록 하였습니다.
> 개인적으로 클래스 내부에서 인스턴스를 담고 있는 것에 거부감이 드는데 관련한 컨벤션 같은게 혹시 있을까요?

클래스 내부에 인스턴스를 담고있는것이 거부감이 든다면 말씀주신것처럼 respository를 이용한 외부 저장소를 이용하면 됩니다.
참고로 Integer.java 에 IntegerCache 부분을 한번 참고해보시면 좋을 것 같아요.

Reply:
현재 구조에서 좀 더 개선해보자고하면 아래처럼 할 수 있을 것 같아요.
```suggestion
        return Optional.ofNullable(getCacheIfExists(value)).orElseGet(() -> createNewCache(value));
```
range 검증은 createNewCache 여기 안으로 들어가구요.

Reply:
validation을 내부로 넣고 합치는 방법이 있었군요. 좋은 것 같아요. 캐쉬는 좀 더 찾아보고 생각해보는게 좋을 것 같네요.

Reply:
IntegerCache 클래스와 valueOf 메서드에서 이를 사용하는 부분을 읽어봤어요. 좋은 참고자료 감사합니다:)